tx.go

```diff
@@ -440,6 +467,19 @@ func (tx *Tx) WriteTo(w io.Writer) (n int64, err error) {
 	return n, nil
 }
 
+func sameFile(f1, f2 *os.File) (bool, error) {
+	fi1, err := f1.Stat()
+	if err != nil {
+		return false, fmt.Errorf("failed to get fileInfo of the first file: %w", err)
```

Comment:
Would logging the `f1` or its name be relevant ?

Reply:
Actually the two files have the same name. But added the filename into the log anyway, thx